python/pyarrow/_dataset.pyx

```diff
@@ -75,6 +70,59 @@ def _get_orc_fileformat():
     return _orc_fileformat
 
 
+_dataset_pq = False
+
+
+def _get_parquet_classes():
+    """
+    Import Parquet class files on first usage (to avoid circular import issue
+    when `pyarrow._dataset_parquet` would be imported first)
+    """
+    global _dataset_pq
+    if _dataset_pq is False:
+        try:
+            import pyarrow._dataset_parquet as _dataset_pq
+        except ImportError:
+            _dataset_pq = None
+
+
+def _get_parquet_symbol(name):
+    _get_parquet_classes()
+    return _dataset_pq and getattr(_dataset_pq, name)
+
+
+def _get_parquet_fileformat():
+    """
+    Import ParquetFileFormat on first usage (to avoid circular import issue
+    when `pyarrow._dataset_parquet` would be imported first)
+    """
+    return _get_parquet_symbol('ParquetFileFormat')
```

Comment:
Since those functions are each only used once, _could_ also use `_get_parquet_symbol("ParquetFileFormat")` directly below instead of the `_get_parquet_fileformat()` indirection

Reply:
Ah, perhaps indeed.